### DISCUSSION TITLE
How can I force the pandas table to be \textwidth or specific size.

### DISCUSSION THREAD

**Discussion by mahdikooshkbaghi** [quarto-dev/quarto-cli#5855] [Q&A] [answered]
### Description

Hi I have the following cell in my `jupyter-notebook` which read the air quality of NYC from public dataset.
The dataframe is wide and when I use `quarto` to generate `a4` size paper, half of the paper is out of the margin.
Is there any possible option I can add to my `_quarto.yml` to handle this scenario. In `LaTeX` usually `\resizebox{\textwidth}` is the to go method for me for any wide tables but here I am not sure how to force it.
Thank you all.

```python
import pandas as pd
# Global parameters
GEO_NAME = "Bronx"
SUBSTANCE_NAME = "Ozone (O3)"
AIR_QUALITY_LINK = "https://data.cityofnewyork.us/api/views/c3uy-2p5r/rows.csv?accessType=DOWNLOAD"
# Read data
df = pd.read_csv(AIR_QUALITY_LINK)
df.head(10)
```

**Comment by mcanouil** [comment]
Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by mahdikooshkbaghi** [reply]
Thanks, 
I made a repo containing a minimal example of `jupyter-notebook` with wide table and the output pdf produced by `quarto`.

https://github.com/mahdikooshkbaghi/quarto_wide_table/tree/main

**Reply by mcanouil** [reply]
Thanks, I'll take a look.
Meanwhile, usually the strategy for such cases is to have the code cell output the LaTex code surrounding the table you are generating.

**Reply by mcanouil** [reply, marked as answer]
The only solution is to decrease font size, or to avoid having wide tables in the first place.
There is not auto-resize in LaTeX.
You have to do that yourself.
See an example (and one way amongst others) below.

````qmd
---
title: "Quarto Playground"
format: pdf
---


`{\tiny`{=latex}

```{python}
#| echo: false
import pandas as pd
# Global parameters
GEO_NAME = "Bronx"
SUBSTANCE_NAME = "Ozone (O3)"
AIR_QUALITY_LINK = "https://data.cityofnewyork.us/api/views/c3uy-2p5r/rows.csv?accessType=DOWNLOAD"
# Read data
df = pd.read_csv(AIR_QUALITY_LINK)
df.head(10)
```

`}`{=latex}
````
